lib/include/irritator/core.hpp

```diff
@@ -5220,14 +5220,76 @@ struct constant
     }
 };
 
-struct filter
-{
+struct filter {
     port x[1];
     port y[1];
-    time sigma = time_domain<time>::infinity;
-    double current_value = 0.0;
-    double up_threshold = 0.0;
-    double down_threshold = 0.0;
+    time sigma;
+
+    double default_lower_threshold;
+    double default_upper_threshold;
+
+    double lower_threshold;
+    double upper_threshold;
+
+    double default_value;
+    double default_input_coeff;
+
+    double value;
```

Comment:
Store a message instead of a double. double is for QSS1 but QSS2 send two real and QSS3 send three real.

Reply:
Please what's the data type of a message? I don't understand!!!

Reply:
`irt::message` : https://github.com/vle-forge/irritator/blob/48ae536b9dfcd3647c0f3d62f7799d19a47c22ba/lib/include/irritator/core.hpp#L783
`real[0]` : value
`real[1]` : first derivative
`real[2]` : second derivative

Store `irt::message` instead of a `double`.